src/components/Avatar/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import StyledAvatar from './styled/avatar';
+import StyledImage from './styled/image';
+import StyledText from './styled/text';
+import StyledIcon from './styled/icon';
+import RenderIf from '../RenderIf';
+import { BaseProps } from '../types';
+
+type AvatarSizes = 'large' | 'medium' | 'small' | 'x-small';
+
+interface Props extends BaseProps {
+    icon?: ReactNode;
+    initial?: string;
+    src?: string;
+    size?: AvatarSizes;
+    style?: object;
+}
+
+const Avatar: React.FC<Props> = props => {
+    const { icon, initial, src, size, style } = props;
+    const isImage = !!src;
+    const isIcon = !!(icon !== null && src === '');
```

Comment:
The icon has the last priority, so if initials are passed the component should render the initials and not the icon. Keep in mind that only one thing should be rendered.

Reply:
gotcha!